wifi-subsys/components/tools/test/test_cbor.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022 Mesh4all <mesh4all.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @brief   Non Volatile Storage
+ *
+ * @author  eduazocar <eduardo@turpialdev.com>
+ */
+#include <stdio.h>
+#include <stdint.h>
+#include "esp_log.h"
+#include "unity.h"
+#include "serialization.h"
+
+static char* key_element[] = {"Key test", "Key_1", "Secret_key"};
+size_t len;
+uint8_t cbor_code[20];
+int64_t val_received;
+
+TEST_CASE("Encode the int value = 32", "cbor")
+{
+    esp_err_t err = encode(key_element[0], 34, cbor_code, &len);
+
+    if (err != ESP_OK)
+    {
+        TEST_FAIL();
+    }
+}
+
+TEST_CASE("Decode the int value that belongs to key_element[0] ", "cbor")
+{
+    esp_err_t err = decode((uint8_t*)cbor_code, key_element[0], &val_received, len);
+
+    if (err != ESP_OK || val_received != 34)  // val is int received and 34 is data origin
```

Comment:
I recommend separating the if, in this way it would work, but in this way you would not identify what error you are getting.